proxy/src/telemetry/metrics/mod.rs

```diff
@@ -597,14 +627,107 @@ impl HyperService for Serve {
                 .with_status(StatusCode::NotFound));
         }
 
-        let body = {
-            let metrics = self.metrics.lock()
-                .expect("metrics lock poisoned");
-            format!("{}", *metrics)
+        let metrics = self.metrics.lock()
+            .expect("metrics lock poisoned")
+            .deref()
+            .clone();
+
+        let is_gzip = is_gzip(&req);
```

Comment:
Please add some comments about how this works in terms of memory usage and Content-Length. In particular, document whether this buffers the entire output in memory instead of streaming it, or is it intended to stream it?

If we're buffering the entire output in memory then let's document the rationale; i.e. that we expect the output will always be small, i.e. less than 100K, and some justification for that. Maybe we should have a TODO comment to switch to a streaming implementation in this case.

If we're streaming the response body then let's document that we're streaming the response body because we don't want to allocate a potentially large amount of memory.